Socket work

mitch-gilmore opened the pull request:
This PR separates the ML and Socket work into a separate PRs to make reviews faster.

dayannat committed:
Socket programs  to transfer data line by line

dayannat committed:
Merge remote-tracking branch 'upstream/main'

dayannat committed:
Attempt to generalize sockets

This code is attempting to receive data directly from the python CLI program that simulates board data

dayannat committed:
Attempt to generalize sockets

This code makes attempts to grab the data from the CLI program that simulates board data

dayannat committed:
Merge branch 'main' of https://github.com/dayannat/brain-4ce

dayannat committed:
Fixed errors

mitch-gilmore committed:
mergeing in socket work

mitch-gilmore committed:
removed 'Unused File'

jonathanm-12 commented:
Code looks fine as it uses a simple TCP socket technique. The server_socket.py file serves as an effective intermediary between the three Python programs in our project. I pray to God that this won't be the last push request I approve.

jonathanm-12 reviewed (dismissed):
Code looks fine as it uses a simple TCP socket technique. The server_socket.py file serves as an effective intermediary between the three Python programs in our project.

mitch-gilmore commented:
@dayannat @jonathanm-12 Is the `server_socket.py` file needed or just written for the latest demo?

jonathanm-12 commented:
> @dayannat @jonathanm-12 Is the `server_socket.py` file needed or just written for the latest demo?

it will be needed

@mgilmore42

mitch-gilmore commented:
@jonathanm-12 

I guess I'm not understanding its utility. Can you please expand? My impression was the clients/servers would need to be embedded into the 3 parts (boardStreamer, ML, Virtual Environment). Like so..

boardStreamer ----(TCP)---> ML ----(TCP)---> Virtual Environment

Where boardStreamer serves to ML. ML is a client of boardStreamer and a server of Virtual Environment. And Virtual Environment is purely a client of ML. Similar to what was added to `boardStreamer/__main__.py` around line 100.

1. I this intuition incorrect?
2. If this it true how does an additional script fit into the pipeline?

To me it looks like just a demo to capture the data and acts purely as a client (against its name). Please explain where my understanding of its utility diverges from your understanding or how my understanding can be reconciled with your own.

mitch-gilmore committed:
removed server_socket.py

mitch-gilmore committed:
changed to assign port number based on CLI args

mitch-gilmore commented:
@jonathanm-12 I removed the file for now until it utility can be better explained/defended. I also changed the port assignment to use the command-line argument instead of being hard coded.

mitch-gilmore committed:
Merge branch 'BostonUniversitySeniorDesign:main' into socket_work

jonathanm-12 commented:
Will approve once I see that this can work in sync with ML and virtual environment.

dayannat commented:
> @jonathanm-12
> 
> I guess I'm not understanding its utility. Can you please expand? My impression was the clients/servers would need to be embedded into the 3 parts (boardStreamer, ML, Virtual Environment). Like so..
> 
> boardStreamer ----(TCP)---> ML ----(TCP)---> Virtual Environment
> 
> Where boardStreamer serves to ML. ML is a client of boardStreamer and a server of Virtual Environment. And Virtual Environment is purely a client of ML. Similar to what was added to `boardStreamer/__main__.py` around line 100.
> 
> 1. I this intuition incorrect?
> 2. If this it true how does an additional script fit into the pipeline?
> 
> To me it looks like just a demo to capture the data and acts purely as a client (against its name). Please explain where my understanding of its utility diverges from your understanding or how my understanding can be reconciled with your own.

This was meant to be a demo for the final prototype testing. As there was no ML model existing at the time, there could be no closed connection. Therefore, the purpose was to prove functionality of the socket programming. From here, what's left is to add sockets through the ML model that will receive data from the board and output the predictive results to the virtual environment.
